core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/upgrade/validation/handler/ClusterUpgradeDiskSpaceValidationHandler.java

```diff
@@ -32,13 +33,22 @@ public class ClusterUpgradeDiskSpaceValidationHandler extends ExceptionCatcherEv
     @Inject
     private StackService stackService;
 
+    @Inject
+    private ParcelService parcelService;
+
     @Override
     protected Selectable doAccept(HandlerEvent<ClusterUpgradeDiskSpaceValidationEvent> event) {
-        LOGGER.debug("Accepting Cluster upgrade validation event.");
+        LOGGER.debug("Accepting Cluster upgrade validation event {}", event);
         ClusterUpgradeDiskSpaceValidationEvent request = event.getData();
         Long stackId = request.getResourceId();
         try {
-            diskSpaceValidationService.validateFreeSpaceForUpgrade(getStack(stackId), request.getRequiredFreeSpace());
+            Stack stack = stackService.getByIdWithListsInTransaction(stackId);
+            if (request.isUpgradePreparation()) {
+                LOGGER.debug("Removing the unused parcels from the cluster upgrade ");
+                parcelService.removeUnusedParcelComponents(stack);
```

Comment:
it's strange a validator removes stuff
either this class should be renamed or this logic moved to somewhere else